service-statistics-common/src/main/java/fi/nls/oskari/control/statistics/db/RegionSet.java

```diff
@@ -94,32 +67,51 @@ public String getAttributes() {
     public void setAttributes(String attributes) {
         this.attributes = attributes;
     }
-    
-    public String getUrl() {
-        if (url == null || url.trim().equals("")) {
-            // Default value.
-            return "http://localhost:8080/geoserver";
-        }
-        return url;
+
+    public JSONObject asJSON() {
+        return JSONHelper.createJSONObject("regionIdTag", getIdProperty());
     }
 
-    public void setUrl(String url) {
-        this.url = url;
+    public String getNameProperty() {
+        return getStatsJSON().optString("nameIdTag");
     }
 
-    public String getType() {
-        return type;
+    public String getIdProperty() {
+        return getStatsJSON().optString("regionIdTag");
     }
-    
-    public void setType(String type) {
-        this.type = type;
+
+    public String getFeaturesUrl() {
+        return getStatsJSON().optString("featuresUrl");
     }
 
-    public String getSrs() {
-        return srs;
+    private JSONObject getStatsJSON() {
+        if (stats == null) {
+            JSONObject json = JSONHelper.createJSONObject(attributes);
+            if (json != null) {
+                stats = json.optJSONObject("statistics");
+            }
+            if (stats == null) {
+                stats = new JSONObject();
+            }
+        }
+        return stats;
     }
 
-    public void setSrs(String srs) {
-        this.srs = srs;
+    public List<Region> getRegions(String requestedSRS) throws IOException, ServiceException {
```

Comment:
Not a big fan of this change since now Regionset is no longer a value object, but contains logic for fetching it's regions. But meh... let's roll with this.

Reply:
That's fair. Will re-evaluate this soon.